go-controller/pkg/cni/cni.go

```diff
@@ -148,12 +183,45 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, clientset *ClientSet) (*Resp
 
 	podInterfaceInfo.SkipIPConfig = kubevirt.IsPodLiveMigratable(pod)
 
+	var userDefinedNetPodInfo *PodInterfaceInfo
+	if hasUserDefinedPrimaryNetwork {
+		klog.Infof("About to plumb the user defined NET for %q on pod %q", userDefinedNetworkName, podName)
+		userDefinedNetPodInfo, err = PodAnnotation2PodInfo(
+			annotations,
+			userDefinedPrimaryNetworkAnnotation,
+			pr.PodUID,
+			"",
+			userDefinedNetworkName, // TODO: we need to find a way to have here access to the network name
+			userDefinedNetworkName, // TODO: we can have the NAD name from the annotations. But the network name is missing. maybe we can add that to the annotations
+			pr.CNIConf.MTU,
```

Comment:
we need a way to grab the MTU from the **right** network.

I think we need to actually read the NAD on ovnkube-controller.

Reply:
right now I think the best way for @tssurya 's helper function to just spit out the NAD object.

Then anyone that needs it, can access the metadata.name info / unmarshall the NAD.Spec.Config data into an OVN-Kubernetes configuration object and do whatever they need. 

Food for thought.